com.unity.render-pipelines.universal/Editor/Converter/Converters/RenderSettingsConverter.cs

```diff
@@ -0,0 +1,301 @@
+                if (GetEquivalentRenderMode(m_GraphicsTierSettings.RenderingPath) == RenderingMode.Deferred)
+                    defaultIndex = m_RenderingModes.IndexOf(RenderingMode.Deferred);
+            }
+
+            asset.m_RendererDataList = renderers.ToArray();
+            asset.m_DefaultRendererIndex = defaultIndex;
+
+            // Create Pipeline asset on disk
+            AssetDatabase.CreateAsset(asset, path);
+            // Assign asset
+            QualitySettings.SetQualityLevel(settings.Index);
+            QualitySettings.renderPipeline = asset;
+
+            // return to original quality level
+            QualitySettings.SetQualityLevel(currentQualityLevel);
+        }
+
+        private ScriptableRendererData CreateRendererDataAsset(string assetPath, RenderingPath renderingPath,
+            string fileName)
+        {
+            var rendererAsset =
+                UniversalRenderPipelineAsset.CreateRendererAsset(assetPath, RendererType.UniversalRenderer, true, fileName)
+                as UniversalRendererData;
+            //Missing API to set deferred or forward
+            rendererAsset.renderingMode =
+                renderingPath == RenderingPath.Forward ? RenderingMode.Forward : RenderingMode.Deferred;
+            //missing API to assign to pipeline asset
+            return rendererAsset;
+        }
+
+        /// <summary>
+        /// Sets all relevant RP settings in order they appear in URP
+        /// </summary>
+        /// <param name="asset">Pipeline asset to set</param>
+        /// <param name="settings">The ProjectSettingItem with stored settings</param>
+        private void SetPipelineSettings(UniversalRenderPipelineAsset asset, RenderSettingItem settings)
+        {
+            // General
+            asset.supportsCameraDepthTexture = settings.SoftParticles;
+
+            // Quality
+            asset.supportsHDR = m_GraphicsTierSettings.HDR;
+            asset.msaaSampleCount = settings.MSAA == 0 ? 1 : settings.MSAA;
+
+            // Main Light
+            asset.mainLightRenderingMode = settings.PixelLightCount == 0
+                ? LightRenderingMode.Disabled
+                : LightRenderingMode.PerPixel;
+            asset.supportsMainLightShadows = settings.Shadows != ShadowQuality.Disable;
+            asset.mainLightShadowmapResolution =
+                GetEquivalentMainlightShadowResolution((int)settings.ShadowResolution);
+
+            // Additional Lights
+            asset.additionalLightsRenderingMode = settings.PixelLightCount <= 1
+                ? LightRenderingMode.Disabled
+                : LightRenderingMode.PerPixel;
+            asset.maxAdditionalLightsCount = Mathf.Max(0, settings.PixelLightCount - 1);
+            asset.supportsAdditionalLightShadows = settings.Shadows != ShadowQuality.Disable;
+            asset.additionalLightsShadowmapResolution =
+                GetEquivalentAdditionalLightAtlasShadowResolution((int)settings.ShadowResolution);
```

Comment:
This completely disables lights that should probably stay enabled. On Low settings in VikingVillage with Pixel Light Count = 0, lights are still enabled in built-in but set to per-vert. Completely disabling them might be confusing to users, especially with the main light or when deferred rendering is used.

Reply:
Another question on this same block is when shadows are set to Hard Only in built-in, should we completely disable shadows in URP or set them to really low resolution? Maybe do this for only the Main Light at least?

Reply:
Have tweaked the pixel light count logic to be more in line with builtin, if 0 lights, main light is still pixel, and I set 4 vertex lights for additional. If any number of pixel lights, these are only counted as Additional lights, so ignoring the main light which builtin seems to also do.

As for soft shadows, this is handled further down, pretty much if builtin supported 'All' I enable soft shadows, builtin it was more per light, but I think in URP we will be moving towards having a unified shadowing across all lights.